src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_INameOfOperation.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public partial class IOperationTests : SemanticModelTestBase
+    {
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void NameOfFlow_01()
+        {
+            string source = @"
+class C
+{
+    void M(bool b, int i1, int i2)
+    /*<bind>*/{
+        string test = nameof(b ? i1 : i2);
+    }/*</bind>*/
+}
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS8081: Expression does not have a name.
+                //         string test = nameof(b ? i1 : i2);
+                Diagnostic(ErrorCode.ERR_ExpressionHasNoName, "b ? i1 : i2").WithLocation(6, 30),
+                // CS0219: The variable 'test' is assigned but its value is never used
```

Comment:
>// CS0219: The variable 'test' is assigned but its value is never used [](start = 16, length = 70)

Also creates diagnostics like that, which is not a problem when there are other errors, but could be annoying for success cases